### PR TITLE
Add Falcon benchmarking.

### PR DESCRIPTION
**Pull request by anhu**

### CODE REVIEW

File: wolfcrypt/benchmark/include.am [hunk 1]
```diff
@@ -9,6 +9,8 @@ wolfcrypt_benchmark_benchmark_SOURCES      = wolfcrypt/benchmark/benchmark.c
 wolfcrypt_benchmark_benchmark_LDADD        = src/libwolfssl.la $(LIB_STATIC_ADD)
 wolfcrypt_benchmark_benchmark_DEPENDENCIES = src/libwolfssl.la
 noinst_HEADERS += wolfcrypt/benchmark/benchmark.h
+noinst_HEADERS += wolfcrypt/benchmark/falcon_level1_key.h
```
Comment (dgarske) [inline wolfcrypt/benchmark/include.am:12]: These should be in ./certs and update `gencertbuf.pl` to populate the wolfssl/certs_test.h.

File: wolfcrypt/benchmark/benchmark.c [hunk 2]
```diff
@@ -198,6 +198,11 @@
 #ifdef WOLFCRYPT_HAVE_SAKKE
     #include <wolfssl/wolfcrypt/sakke.h>
 #endif
+#ifdef HAVE_LIBOQS
+    #include <wolfssl/wolfcrypt/falcon.h>
+    #include "falcon_level1_key.h"
```
Comment (dgarske) [inline wolfcrypt/benchmark/benchmark.c:203]: Would prefer having a macro in `wolfssl/certs_test.h` updated using `gencertbuf.pl`. Use a new macro with `HAVE_LIBOQS` in there...

File: wolfcrypt/benchmark/benchmark.c [hunk 3]
```diff
@@ -6404,6 +6422,97 @@ void bench_sakke(void)
...
+#ifdef HAVE_LIBOQS
+void bench_falconKeySign(byte level)
+{
+    int    ret = 0;
+    falcon_key key;
+    double start;
+    int    i, count;
+    byte   sig1[FALCON_LEVEL1_SIG_SIZE];
+    byte   sig5[FALCON_LEVEL5_SIG_SIZE];
+    byte   msg[512];
```
Comment (dgarske) [inline wolfcrypt/benchmark/benchmark.c:6434]: Do we have a macro now for `sig` max size?

File: wolfcrypt/benchmark/benchmark.c [hunk 4]
```diff
@@ -6404,6 +6422,97 @@ void bench_sakke(void)
...
+    const char**desc = bench_desc_words[lng_index];
+
+    ret = wc_falcon_init(&key);
+    if (ret != 0) {
+        printf("wc_falcon_init failed\n");
+        return;
+    }
+
+    ret = wc_falcon_set_level(&key, level);
+    if (ret != 0) {
```
Comment (dgarske) [inline wolfcrypt/benchmark/benchmark.c:6445]: The code needs refactored to make sure `wc_falcon_free(&key);` will always be called after `wc_falcon_init` succeeds.

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfcrypt/benchmark/benchmark.c (hunk 3 above)
Comment (anhu) [reply wolfcrypt/benchmark/benchmark.c:6434]: We do, but by doing this I get to use sizeof() below which looks neater to me.

File: wolfcrypt/benchmark/benchmark.c (hunk 4 above)
Comment (anhu) [reply wolfcrypt/benchmark/benchmark.c:6445]: Memory management bites me again! Will do it.

File: wolfcrypt/benchmark/benchmark.c (hunk 2 above)
Comment (anhu) [reply wolfcrypt/benchmark/benchmark.c:203]: Will fix.

File: wolfcrypt/benchmark/include.am (hunk 1 above)
Comment (anhu) [reply wolfcrypt/benchmark/include.am:12]: Will fix.

File: wolfcrypt/benchmark/benchmark.c (hunk 3 above)
Comment (anhu) [reply wolfcrypt/benchmark/benchmark.c:6434]: yeah...alright...I'll change.

File: wolfcrypt/benchmark/benchmark.c (hunk 4 above)
Comment (anhu) [reply wolfcrypt/benchmark/benchmark.c:6445]: oh right...I remember now...I got careless because I thought this is just benchmarking code.

**Review by dgarske — APPROVED** [review]

**Pull request merged**